### PR TITLE
Steam: Fix tag naming

### PR DESCRIPTION
**Pull request by Jeshibu**
Turns out the existing mechanism to download a new set of tags always broke, this is testable for games with the tag "Fox" right now. This fixes that and also employs a newly discovered way to download tags from the Steam API instead of scraping HTML.

### CODE REVIEW

File: source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs [hunk 1]
```diff
@@ -46,48 +47,88 @@ private string GetPackedTagNameFilePath()
...
         {
-            string url = $"https://store.steampowered.com/search/?l={settings?.LanguageKey}";
+            var existingFile = GetFileContents();
+
+            string url = $"https://api.steampowered.com/IStoreService/GetTagList/v1?language={settings?.LanguageKey}&have_version_hash={existingFile?.response?.version_hash}";
+
             logger.Debug("Downloading " + url);
-            var content = downloader.DownloadString(url);
-            var matches = tagNameRegex.Matches(content);
+            var httpClient = new HttpClient();
```
Comment (JosefNemec) [inline source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs:92]: Why this change? You are creating new HttpClient every time this is called, which is bad (not even disposing it after use). What was wrong with the original reused downloader method?

File: source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs [hunk 2]
```diff
@@ -46,48 +47,88 @@ private string GetPackedTagNameFilePath()
...
+            downloadTask.Wait();
+            var downloadResult = downloadTask.Result;
+            var contentTask = downloadResult.Content.ReadAsStringAsync();
+            contentTask.Wait();
+
+            var filePath = GetTagNameFilePath();
 
-            var output = new Dictionary<int, string>();
-            foreach (var match in matches.Cast<Match>())
+            if (downloadResult.StatusCode != System.Net.HttpStatusCode.NotModified && !string.IsNullOrWhiteSpace(contentTask.Result))
```
Comment (JosefNemec) [inline source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs:101]: I'm not sure if this is necessary if you already downloaded all that data anyways.

File: source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs (hunk 1 above)
Comment (Jeshibu) [reply source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs:92]: Because I need to check if it's a 304 result (no change, no result body). Could change it to just always download the file and only check if the content is empty. That's probably better, right?

Comment (JosefNemec) [reply source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs:92]: You are calling GET so you are downloading the body always either way. But the main point is that you should be reusing HttpClient instances, not creating new ones every time this is created.

If you want to check response code before downloading the file, you can use `HttpDownloader.GetResponseCode`, it sends HEAD request and gives you response code and headers without downloading body. Then you could just use the original downloader method.

File: source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs (hunk 2 above)
Comment (Jeshibu) [reply source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs:101]: The response body is empty when it's NotModified, if that's what you mean

File: source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs (hunk 1 above)
Comment (Jeshibu) [reply source/Libraries/SteamLibrary/SteamShared/SteamTagNamer.cs]: Removed HttpClient usage now
